src/main/java/org/opensearch/security/user/InternalUserProvider.java

```diff
@@ -0,0 +1,20 @@
+package org.opensearch.security.user;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.IOException;
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+
+/**
+ * This interface will be defined in core following these changes. For now this is used for testing in the Security Plugin.
+ *
+ */
+public interface InternalUserProvider {
+
+    public void putInternalUser(String userInfo) throws java.io.IOException;
```

Comment:
I'd recommend against throwing IOException, can we try /catch where this is coming from or if you think its needed throw specific exceptions?